Kudu.Core/Deployment/FetchDeploymentManager.cs

```diff
@@ -222,6 +225,12 @@ public class FetchDeploymentManager : IFetchDeploymentManager
                             // Here, we don't need to update the working files, since we know Fetch left them in the correct state
                             // unless for GenericHandler where specific commitId is specified
                             bool deploySpecificCommitId = !String.IsNullOrEmpty(deploymentInfo.CommitId);
+                            if (PostDeploymentHelper.IsAzureEnvironment() && deploymentInfo.FixedDeploymentId != null)
```

Comment:
Don't we want to handle 404 when invoking /api/app/updatedeploystatus? We may see SCM-DEPLOYMENT-ID coming from an FE instance that has been upgraded, but this call may hit an FE instance that hasn't been upgraded right?

Reply:
@purva-vasudeo

Reply:
Yes I think we should. In case of a 404 though, should we retry the request or just ignore the 404 and go ahead with the code? @sanchitmehta

Reply:
Silently ignoring the 404 is good enough. The only purpose is to avoid regressing existing deployment pipelines, etc during the brief upgrade period.